inst/@sym/assume.m

```diff
@@ -116,47 +116,48 @@
 %% @seealso{@@sym/assumeAlso, assumptions, sym, syms}
 %% @end deftypemethod
 
-%% Author: Colin B. Macdonald
-%% Keywords: symbolic
 
-function varargout = assume(x, varargin)
+function varargout = assume(xx, varargin)
 
   assert (nargin > 1, 'General algebraic assumptions are not supported');
 
-  xstr = x.flat;
+  for i = 1:length (xx)
+    x = subsref (xx, substruct('()', {i}));
+    xstr = x.flat;
 
-  if (nargin > 1 && strcmp(varargin{1}, 'clear'))
-    assert (nargin == 2, 'assume: clear cannot be combined with other assumptions')
-    newx = sym(xstr);
-  else
-    for n = 2:nargin
-      cond = varargin{n-1};
-      ca.(cond) = true;
+    if (nargin > 1 && strcmp(varargin{1}, 'clear'))
+      assert (nargin == 2, 'assume: clear cannot be combined with other assumptions')
+      newx = sym(xstr);
+    else
+      for n = 2:nargin
+        cond = varargin{n-1};
+        ca.(cond) = true;
+      end
+      newx = sym(xstr, ca);
     end
-    newx = sym(xstr, ca);
-  end
 
-  if (nargout > 0)
-    varargout{1} = newx;
-    return
-  end
+    if (nargout > 0)
+      varargout{i} = newx;
```

Comment:
little question here, if we do
```
syms x
f=x*2;
x = assume(x, 'positive')
```
```f``` will be updated with the new assumptions?

if not will be updated i think is missed the ```return```, and if not is fine.

(This for assume and assumeAlso)

Thx. Cya.